### PR TITLE
Removal of Redundant Features in Libraries #minor

### PR DESCRIPTION
**Pull request by anilbey**
#### Context:

This PR removes unused or untested features from active maintenance, aiming to cut down on the number of features we have to maintain and to make sure the ones we do keep are consistent in dealing with missing values, exceptions, and other edge cases that are currently compromising our scientific results.
It addresses the ongoing work documented in [Issue #321](https://github.com/BlueBrain/eFEL/issues/321).

#### Modifications:

1. **Removing LibV4:**
    - `LibV4` has a single function called `peak_indices`. `LibV5` already has an improved version of it.
2. **Keeping Only `depolarized_base` in LibV3:**
    - Only that feature is used and tested. 
    - Rest of the features already exists in LibV5 and LibV1 (that are used and tested).
3. **Additional Removals and Replacements:**
   - `LibV1::peak_indices` is removed, as the `LibV5` implementation is in use.
   - ~~`LibV2::E6-E40` protocols are neither utilised nor tested; the membrane systems team uses the `LibV7` implementation.~~
   - `LibV1::spike_width1` is discarded, as `LibV5::spike_half_width` is in use.
   - `LibV1::rest_voltage_value` is removed, as `LibV5::voltage_base` is in use.
   - `LibV1::min_AHP_values` is removed as `LibV5::min_AHP_values` is in use.
   - `LibV1::AHP_depth_abs` is removed, as `LibV5::AHP_depth_abs` is in use.
   - `LibV1::min_AHP_indices` is removed, as `LibV5::min_AHP_indices` is in use.

#### Note:

While these features and functions are removed from active maintenance, it's important to note that their implementations remain accessible within the version control system, preserving the ability to revisit or reactivate them in future developments.

### CODE REVIEW

**Comment by AurelienJaquier** [comment]
Shouldn't we wait until we have E features merged from LibV7 before removing them?

File: tests/test_basic.py
Comment (AurelienJaquier) [inline tests/test_basic.py:1438]: Shouldn't we just remove this test altogether with the DependencyV5_LibV4peakindices.txt file?
It is testing features that are no longer present.

**Comment by anilbey** [comment]
> Shouldn't we wait until we have E features merged from LibV7 before removing them?

Good point. Since there were also no tests for the E features I removed them in the PR but we can wait actually. Shall I revert that single commit?

File: tests/test_basic.py
Comment (anilbey) [reply tests/test_basic.py]: Ok I deleted that and added another test for `setDependencyFileLocation` since we need a test for checking the case when `setDependencyFileLocation` works.

**Review by AurelienJaquier — APPROVED** [review]
Great! Looks good to me!

**Comment by anilbey** [comment]
Alright merging... :rocket:  
With this change, there are no dead-code C++ feature implementations anymore. Each implemented C++ feature can now be called from the Python API.

**Pull request merged**
